### PR TITLE
Add build, tag for oniguruma, remove for mysql_go_regex, update README

### PR DESCRIPTION
**Pull request by juanjux**
- Adds a build tag for oniguruma.
- Removes the existing one for Go's regex.
- Updates the README removing references to oniguruma.

Signed-off-by: Juanjo Alvarez <juanjo@sourced.tech>

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -69,6 +69,8 @@ github.com/opentracing/opentracing-go v1.0.2 h1:3jA2P6O1F9UOrWVpwrIo17pu01KWvNWg
 github.com/opentracing/opentracing-go v1.0.2/go.mod h1:UkNAQd3GIcIGf0SeVgPpRdFStlNbqXla1AfSYxPUl2o=
 github.com/pascaldekloe/goe v0.0.0-20180627143212-57f6aae5913c h1:Lgl0gzECD8GnQ5QCWA8o6BtfL6mDH5rQgM4/fX3avOs=
 github.com/pascaldekloe/goe v0.0.0-20180627143212-57f6aae5913c/go.mod h1:lzWF7FIEvWOWxwDKqyGYQf6ZUaNfKdP144TG7ZOy1lc=
+github.com/pbnjay/memory v0.0.0-20190104145345-974d429e7ae4 h1:MfIUBZ1bz7TgvQLVa/yPJZOGeKEgs6eTKUjz3zB4B+U=
```
Comment (kuba--) [inline go.sum]: what is that?

File: internal/regex/regex_oniguruma.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// +build mysql_go_onigurama
```
Comment (kuba--) [inline internal/regex/regex_oniguruma.go]: enry has just `oniguruma` afaik

File: go.sum (hunk 1 above)
Comment (juanjux) [reply go.sum]: Seems `go mod tidy` added it.

File: internal/regex/regex_oniguruma.go (hunk 2 above)
Comment (juanjux) [reply internal/regex/regex_oniguruma.go]: It was for symmetry with the previous tag, but will change it to onigurama.

**Comment by ajnavarro** [comment]
@juanjux it is `oniguruma` not `onigurama`

File: internal/regex/regex_oniguruma.go (hunk 2 above)
Comment (juanjux) [reply internal/regex/regex_oniguruma.go]: PS: done.

**Comment by juanjux** [comment]
@ajnavarro renamed...

File: go.sum (hunk 1 above)
Comment (juanjux) [reply go.sum]:
```bash
go mod why github.com/pbnjay/memory 
go: finding github.com/uber/jaeger-lib/metrics/testutils latest
go: finding github.com/uber/jaeger-lib/metrics latest
# github.com/pbnjay/memory
gopkg.in/src-d/go-mysql-server.v0/sql/plan
github.com/pbnjay/memory
```

**Pull request merged**